### PR TITLE
Views with type erasure and benchmarks

### PR DESCRIPTION
**Pull request by h-2**
~~blocked by ericniebler/range-v3#1092~~

----

* Adds new views: view::all, view::drop, view::slice
* Our view::all, view::take, view::drop and view::slice type-erase views when possible (string to string_view and contiguous to std::span).
* Adds benchmarks to everything

### CODE REVIEW

**Comment by h-2** [comment]
Only blocked by the ranges bug, one performance test expected to fail because of that

**Comment by h-2** [comment]
fixed another snipped warning

File: test/performance/range/view/view_drop_view_take_benchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        {
+            for (auto e : c)
+                dummy += e;
+        }
+
+    } else
+    {
+        auto dr = drop_t{}(10);
+        auto ta = take_t{}(10'000);
+        auto v = c | dr | ta | dr | ta | dr | ta;
```
Comment (eseiler) [inline test/performance/range/view/view_drop_view_take_benchmark.cpp]: That results in a size of 9980, doesn't it?

Is this intended?

Now we are kinda comparing the `void, void` which has to do `1'000'000`  accesses to do one iteration vs. the 9980 accesses the view take/drop combination has to do. (And for `take` we were very particular about also using `1'000'000` elements.

File: test/performance/range/view/view_drop_view_take_benchmark.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+        for (auto _ : state)
+        {
+            for (auto e : v)
+                dummy += e;
```
Comment (eseiler) [inline test/performance/range/view/view_drop_view_take_benchmark.cpp]: No random access? 🤔

File: test/performance/range/view/view_all_benchmark.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    [[maybe_unused]] volatile unsigned dummy2 = dummy;
+}
+
+BENCHMARK_TEMPLATE(iterate_left_to_right, std::string, void);
+BENCHMARK_TEMPLATE(iterate_left_to_right, std::string, decltype(std::ranges::view::all));
+BENCHMARK_TEMPLATE(iterate_left_to_right, std::string, decltype(::ranges::view::all));
+BENCHMARK_TEMPLATE(iterate_left_to_right, std::string, decltype(seqan3::view::all));
+
+BENCHMARK_TEMPLATE(iterate_left_to_right, std::vector<int>, void);
```
Comment (eseiler) [inline test/performance/range/view/view_all_benchmark.cpp]: Also here: `int` ?

File: test/performance/range/view/view_take_benchmark.cpp [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+#include <list>
+#include <forward_list>
+#include <string>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/range/view/take.hpp>
+#include <seqan3/range/view/take_exactly.hpp>
+#include <seqan3/range/view/single_pass_input.hpp>
```
Comment (eseiler) [inline test/performance/range/view/view_take_benchmark.cpp]: `s` < `t`

File: test/performance/range/view/view_take_benchmark.cpp [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    unsigned dummy = 0;
+
+    // if single_pass, add view::single_pass_input, otherwise just &
+    using single_t = std::conditional_t<single_pass, decltype(c | view::single_pass_input), container_t &>;
+
+    if constexpr (std::Same<adaptor_t, void>)
+    {
+        for (auto _ : state)
+        {
+            single_t s{c};
```
Comment (eseiler) [inline test/performance/range/view/view_take_benchmark.cpp]: In the other benchmarks you do not benchmark the construction of the view, so you should probably not time this and exclude it via `state.PauseTiming()` and `state.ResumeTiming()` ?

File: test/performance/range/view/view_take_benchmark.cpp [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+        }
+
+    } else
+    {
+        auto adaptor = adaptor_t{}(1'000'000);
+
+        for (auto _ : state)
+        {
+            single_t s{c};
+            auto v = s | adaptor | adaptor | adaptor | adaptor | adaptor
```
Comment (eseiler) [inline test/performance/range/view/view_take_benchmark.cpp]: Same here.

File: test/performance/range/view/view_take_until_benchmark.cpp [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+    if constexpr (std::Same<adaptor_t, void>)
+    {
+        using single_t = std::conditional_t<single_pass, decltype(c | view::single_pass_input), container_t &>;
+
+        for (auto _ : state)
+        {
+            single_t s{c};
+            for (auto e : s)
+                if ((dummy = e) >= 101)
```
Comment (eseiler) [inline test/performance/range/view/view_take_until_benchmark.cpp]: here you do assignment, in the non-void case you do summation

File: include/seqan3/range/view/drop.hpp [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+{
+
+// ============================================================================
+//  drop_fn (adaptor definition)
+// ============================================================================
+
+/*!\brief View adaptor definition for view::drop and view::drop_or_throw.
+ * \tparam or_throw Whether to throw an exception when the input is exhausted before the end is reached.
+ */
+template <bool exactly, bool or_throw>
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/range/view/drop.hpp]: no, it's just a left-over from copy'n'paste

File: include/seqan3/range/view/drop.hpp [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+            return std::ranges::subrange<std::ranges::iterator_t<urng_t>, std::ranges::sentinel_t<urng_t>>{
+                std::move(b), std::ranges::end(urange)};
+        }
+    }
+
+    /*!\brief       Overload for contiguous, sized ranges.
+     * \returns     A std::span over the input.
+     */
+    template <std::ranges::ViewableRange urng_t>
+        requires std::ranges::ContiguousRange<urng_t> && std::ranges::SizedRange<urng_t>
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/range/view/drop.hpp]: Yes, for example:
```cpp
std::vector vec{1, 2, 3, 4, 5, 6};
auto v = vec | view::take_until([] (int i) { return i >= 3; })
```
In that case v is contiguous, but you don't know the size without iterating over it from beginning to end.

File: test/performance/range/view/view_all_benchmark.cpp [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+// ============================================================================
+
+template <typename container_t, typename adaptor_t>
+void iterate_left_to_right(benchmark::State& state)
+{
+    container_t c;
+    c.resize(1'000'000);
+    uint8_t i = 0;
+    for (auto & e : c)
+        e = ++i; // dummy values
```
[the remark this answers is not in the document]
Comment (h-2) [reply test/performance/range/view/view_all_benchmark.cpp]: yes

File: test/performance/range/view/view_all_benchmark.cpp [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+
+    unsigned dummy = 0;
```
[the remark this answers is not in the document]
Comment (h-2) [reply test/performance/range/view/view_all_benchmark.cpp]: hm, well the size is not important her,e but I can change it

File: test/performance/range/view/view_all_benchmark.cpp (hunk 3 above)
Comment (h-2) [reply test/performance/range/view/view_all_benchmark.cpp]: There is no other integer that you can get by writing three letters :wink:

File: test/performance/range/view/view_drop_benchmark.cpp [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+    container_t c;
+    c.resize(1'005'000); // 5'000 more that will be dropped
+    uint8_t i = 0;
+    for (auto & e : c)
+        e = ++i; // dummy values
+
+    std::vector<size_t> access_positions;
+    access_positions.resize(1'000'000);
+    for (size_t i = 0; i < 1'000'000; ++i)
+        access_positions[i] = (((i * 10 + 7) * 3 * i) / 11) % 1'000'000;
```
[the remark this answers is not in the document]
Comment (h-2) [reply test/performance/range/view/view_drop_benchmark.cpp]: It was just something I came up with because I didn't want slow RNG :laughing:

File: test/performance/range/view/view_drop_view_take_benchmark.cpp (hunk 1 above)
Comment (h-2) [reply test/performance/range/view/view_drop_view_take_benchmark.cpp]: good catch

File: test/performance/range/view/view_drop_view_take_benchmark.cpp (hunk 2 above)
Comment (h-2) [reply test/performance/range/view/view_drop_view_take_benchmark.cpp]: :scream:

File: test/performance/range/view/view_take_benchmark.cpp (hunk 5 above)
Comment (h-2) [reply test/performance/range/view/view_take_benchmark.cpp]: I need to recreate the view because of the single_pass case. Since the view creation is constant-time and 1million iterations after that I assume it won't influence it, but I can try with start/stopping, too.

File: test/performance/range/view/view_all_benchmark.cpp (hunk 3 above)
Comment (h-2) [reply test/performance/range/view/view_all_benchmark.cpp]: Actually math operations are fastest for `size_t` so I have changed everything to that (so that the math operations ahve the smallest possible effect).

File: test/performance/range/view/view_all_benchmark.cpp [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+
+    uint8_t dummy = 0;
+
+    if constexpr (std::Same<adaptor_t, void>)
+    {
+        for (auto _ : state)
+        {
+            for (auto e : c)
+                dummy += e;
+        }
```
Comment (eseiler) [inline test/performance/range/view/view_all_benchmark.cpp]:
```cpp
        }
    }
    else
    {
```

?

Also in other files.
If not, at least the new line should go?

File: include/seqan3/range/view/drop.hpp [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+    static std::basic_string_view<char_t, traits_t> impl(std::basic_string<char_t, traits_t, allocator_t> & urange,
+                                                         size_t drop_size)
+    {
+        drop_size = std::min(drop_size, std::ranges::size(urange));
+        return {std::ranges::data(urange) + drop_size, std::ranges::size(urange) - drop_size};
+    }
+
+    //!\overload
+    template <typename char_t, typename traits_t, typename allocator_t>
+    static std::basic_string_view<char_t, traits_t> impl(std::basic_string<char_t, traits_t, allocator_t> const & urange,
```
Comment (eseiler) [inline include/seqan3/range/view/drop.hpp]: Is this line <= 120 characters?

File: include/seqan3/range/view/drop.hpp [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+                                                         size_t drop_size)
+    {
+        drop_size = std::min(drop_size, std::ranges::size(urange));
+        return {std::ranges::data(urange) + drop_size, std::ranges::size(urange) - drop_size};
+    }
+
+    //!\overload
+    template <typename char_t, typename traits_t, typename allocator_t>
+    static std::basic_string_view<char_t, traits_t> impl(std::basic_string<char_t, traits_t, allocator_t> const && urange,
```
Comment (eseiler) [inline include/seqan3/range/view/drop.hpp]: Also here

File: include/seqan3/io/stream/parse_condition_detail.hpp [hunk 16]
```diff
@@ -344,7 +344,7 @@ struct is_in_interval_type : public parse_condition_base<is_in_interval_type<int
     {
         data_t ret{};
 
-        for (uint8_t i = interval_first; i <= interval_last; ++i)
+        for (size_t i = interval_first; i <= static_cast<size_t>(interval_last); ++i)
```
Comment (smehringer) [inline include/seqan3/io/stream/parse_condition_detail.hpp]: why this change?

File: include/seqan3/range/view/drop.hpp [hunk 17]
```diff
@@ -0,0 +1,204 @@
...
+        else
+        {
+            for (size_t i = 0; (i < drop_size) && (b != std::ranges::end(urange)); ++i, ++b);
+
+            return std::ranges::subrange<std::ranges::iterator_t<urng_t>,
+                                         std::ranges::sentinel_t<urng_t>>
+            {
+                std::move(b),
+                std::ranges::end(urange)
+
```
Comment (smehringer) [inline include/seqan3/range/view/drop.hpp]: remove newline :nail_care:

File: include/seqan3/range/view/drop.hpp [hunk 18]
```diff
@@ -0,0 +1,204 @@
...
+            };
+        }
+    }
+
+    /*!\brief       Overload for contiguous, sized ranges.
+     * \returns     A std::span over the input.
+     */
+    template <std::ranges::ViewableRange urng_t>
+        requires std::ranges::ContiguousRange<urng_t> && std::ranges::SizedRange<urng_t>
```
Comment (smehringer) [inline include/seqan3/range/view/drop.hpp]: `//!/cond` missing

File: include/seqan3/range/view/drop.hpp [hunk 19]
```diff
@@ -0,0 +1,204 @@
...
+    {
+        drop_size = std::min(drop_size, std::ranges::size(urange));
+        return std::span{std::ranges::data(urange) + drop_size, std::ranges::size(urange) - drop_size};
+    }
+
+    /*!\brief       Overload for std::basic_string_view.
+     * \returns     A std::span over the input.
+     */
+    template <std::ranges::ViewableRange urng_t>
+        requires std::ranges::ContiguousRange<urng_t> && std::ranges::SizedRange<urng_t> &&
```
Comment (smehringer) [inline include/seqan3/range/view/drop.hpp]: `//!\cond`

File: include/seqan3/range/view/drop.hpp [hunk 20]
```diff
@@ -0,0 +1,204 @@
...
+    {
+        drop_size = std::min(drop_size, std::ranges::size(urange));
+        return urange.substr(drop_size);
+    }
+
+    /*!\brief       Overload for std::basic_string.
+     * \returns     A std::span over the input.
+     */
+    template <typename char_t, typename traits_t, typename allocator_t>
+    static std::basic_string_view<char_t, traits_t> impl(std::basic_string<char_t, traits_t, allocator_t> & urange,
```
Comment (smehringer) [inline include/seqan3/range/view/drop.hpp]: hm it is sufficient here for the overload resolution to give the explicit type basic_string but above you require `is_type_specialisation_of_v<std::basic_string_view>`...?

File: include/seqan3/range/view/drop.hpp [hunk 21]
```diff
@@ -0,0 +1,204 @@
...
+// ============================================================================
+
+namespace seqan3::view
+{
+
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief               A view adaptor that returns all elements after n from the underlying range (or an empty range
```
Comment (smehringer) [inline include/seqan3/range/view/drop.hpp]:
```suggestion
/*!\brief               A view adaptor that returns all elements after drop_size from the underlying range (or an empty range
```
?

File: test/performance/range/view/view_all_benchmark.cpp (hunk 13 above)
Comment (h-2) [reply test/performance/range/view/view_all_benchmark.cpp]: k

File: include/seqan3/io/stream/parse_condition_detail.hpp (hunk 16 above)
Comment (h-2) [reply include/seqan3/io/stream/parse_condition_detail.hpp]: if you have full range (min to max) you overflow and never quit.

File: include/seqan3/range/view/drop.hpp (hunk 20 above)
Comment (h-2) [reply include/seqan3/range/view/drop.hpp]: above I want all four overloads, that's why I use the forwarding reference. Below I want to explicitly forbid taking rvalues and I need different implementations for const and non-const.

File: include/seqan3/range/view/drop.hpp (hunk 21 above)
Comment (h-2) [reply include/seqan3/range/view/drop.hpp]: Hm, I try not to use parameter names in the brief because you see the brief without the rest often. `n` is kind of universally known so I like it for these kind of things...

File: test/performance/range/view/view_drop_benchmark.cpp [hunk 22]
```diff
@@ -0,0 +1,157 @@
...
+
+    if constexpr (std::Same<adaptor_t, void>)
+    {
+        for (auto _ : state)
+        {
+            single_t s{c};
+            for (auto e : s)
+                dummy += e;
+        }
+
```
Comment (eseiler) [inline test/performance/range/view/view_drop_benchmark.cpp]: remove newline 😁

File: test/performance/range/view/view_drop_view_take_benchmark.cpp [hunk 23]
```diff
@@ -0,0 +1,157 @@
...
+
+    if constexpr (std::Same<drop_t, void>)
+    {
+        for (auto _ : state)
+        {
+            single_t s{c};
+            for (auto e : s)
+                dummy += e;
+        }
+
```
Comment (eseiler) [inline test/performance/range/view/view_drop_view_take_benchmark.cpp]: 👀

File: test/performance/range/view/view_take_benchmark.cpp [hunk 24]
```diff
@@ -0,0 +1,178 @@
...
+
+    if constexpr (std::Same<adaptor_t, void>)
+    {
+        for (auto _ : state)
+        {
+            for (auto i : access_positions)
+                dummy += c[i];
+        }
+
+    } else
```
Comment (eseiler) [inline test/performance/range/view/view_take_benchmark.cpp]: 😡

File: test/unit/range/view/view_drop_test.cpp [hunk 25]
```diff
@@ -0,0 +1,183 @@
...
+
+    EXPECT_TRUE(std::ranges::InputRange<decltype(v2)>);
+    EXPECT_FALSE(std::ranges::ForwardRange<decltype(v2)>);
+    EXPECT_FALSE(std::ranges::BidirectionalRange<decltype(v2)>);
+    EXPECT_FALSE(std::ranges::RandomAccessRange<decltype(v2)>);
+    EXPECT_TRUE(std::ranges::View<decltype(v2)>);
+    EXPECT_FALSE(std::ranges::SizedRange<decltype(v2)>);
+    EXPECT_FALSE(std::ranges::CommonRange<decltype(v2)>);
+    EXPECT_TRUE(const_iterable_concept<decltype(v2)>); // because this is always subrange
+//     EXPECT_TRUE((std::ranges::OutputRange<decltype(v2), int>)); //TODO why fail?
```
Comment (smehringer) [inline test/unit/range/view/view_drop_test.cpp]: open TODO

File: include/seqan3/search/algorithm/configuration/max_error.hpp [hunk 26]
```diff
@@ -109,11 +109,11 @@ class max_error : public pipeable_config_element<max_error<errors_t...>, std::ar
         // Only total is set so we set all other errors to the total limit.
         if constexpr (((std::remove_reference_t<errors_t>::_id() == 0) || ...) && sizeof...(errors) == 1)
         {
-            ranges::fill(base_t::value | ranges::view::slice(1, 4), base_t::value[0]);
+            ranges::fill(base_t::value | view::slice(1, 4), base_t::value[0]);
         } // otherwise if total is not set but any other field is set than use total as the sum of all set errors.
         else if constexpr (!((std::remove_reference_t<errors_t>::_id() == 0) || ...) && sizeof...(errors) > 0)
         {
-            base_t::value[0] = std::min(static_cast<uint8_t>(255), ranges::accumulate(base_t::value | ranges::view::slice(1, 4),
+            base_t::value[0] = std::min(static_cast<uint8_t>(255), ranges::accumulate(base_t::value | view::slice(1, 4),
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/search/algorithm/configuration/max_error.hpp]: not yet moved to std and it's not something I added

File: include/seqan3/range/view/drop.hpp [hunk 27]
```diff
@@ -0,0 +1,207 @@
...
+        requires std::ranges::ContiguousRange<urng_t> && std::ranges::SizedRange<urng_t>
+    //!\endcond
+    static auto impl(urng_t && urange, size_t drop_size)
+    {
+        drop_size = std::min(drop_size, std::ranges::size(urange));
+        return std::span{std::ranges::data(urange) + drop_size, std::ranges::size(urange) - drop_size};
+    }
+
+    /*!\brief       Overload for std::basic_string_view.
+     * \returns     A std::span over the input.
```
Comment (rrahn) [inline include/seqan3/range/view/drop.hpp]: noop, a std::basic_string_view

File: include/seqan3/range/view/drop.hpp [hunk 28]
```diff
@@ -0,0 +1,207 @@
...
+                 is_type_specialisation_of_v<std::remove_reference_t<urng_t>, std::basic_string_view>
+    //!\endcond
+    static auto impl(urng_t && urange, size_t drop_size)
+    {
+        drop_size = std::min(drop_size, std::ranges::size(urange));
+        return urange.substr(drop_size);
+    }
+
+    /*!\brief       Overload for std::basic_string.
+     * \returns     A std::span over the input.
```
Comment (rrahn) [inline include/seqan3/range/view/drop.hpp]: std::basic_string_view

File: include/seqan3/range/view/view_all.hpp [hunk 29]
```diff
@@ -0,0 +1,169 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Provides seqan3::view::persist.
```
Comment (rrahn) [inline include/seqan3/range/view/view_all.hpp]: copy'n' paste error

File: include/seqan3/range/view/view_all.hpp [hunk 30]
```diff
@@ -0,0 +1,169 @@
...
+    //!\overload
+    template <typename char_t, typename traits_t, typename allocator_t>
+    static auto impl(std::basic_string<char_t, traits_t, allocator_t> & urange)
+    {
+        return std::basic_string_view<char_t, traits_t>{urange};
+    }
+
+    //!\overload
+    template <typename char_t, typename traits_t, typename allocator_t>
+    static auto impl(std::basic_string<char_t, traits_t, allocator_t> && urange) = delete;
```
Comment (rrahn) [inline include/seqan3/range/view/view_all.hpp]:
```suggestion
    static auto impl(std::basic_string<char_t, traits_t, allocator_t> const && urange) = delete;
```

File: include/seqan3/range/view/view_all.hpp [hunk 31]
```diff
@@ -0,0 +1,169 @@
...
+     */
+    template <std::ranges::ViewableRange urng_t>
+        requires !std::ranges::View<urng_t> && std::ranges::ContiguousRange<urng_t> && std::ranges::SizedRange<urng_t>
+    static auto impl(urng_t && urange)
+    {
+        return std::span{std::ranges::data(urange), std::ranges::size(urange)};
+    }
+
+    /*!\brief       Overload for random-access, sized ranges that aren't views.
+     * \returns     A std::span over the input.
```
Comment (rrahn) [inline include/seqan3/range/view/view_all.hpp]:
```suggestion
     * \returns     A std::ranges::subrange over the input.
```

File: include/seqan3/range/view/view_all.hpp [hunk 32]
```diff
@@ -0,0 +1,169 @@
...
+
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief               A view adaptor that behaves like std::view:all, but type erases contiguous ranges.
+ * \tparam urng_t       The type of the range being processed. See below for requirements. [template parameter is
+ *                      omitted in pipe notation]
+ * \param[in] urange    The range being processed. [parameter is omitted in pipe notation]
+ * \returns             Either `std::basic_string_view{urange}`, or `std::span{urange}`, or `std::view::all(urange)`.
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/range/view/view_all.hpp]: std::view::all is not a type

File: include/seqan3/range/view/view_all.hpp (hunk 30 above)
Comment (h-2) [reply include/seqan3/range/view/view_all.hpp]: good catch 👍

File: include/seqan3/range/view/drop.hpp [hunk 33]
```diff
@@ -0,0 +1,206 @@
...
+public:
+    //!\brief Inherit the base class's Constructors.
+    using base_t::base_t;
+
+private:
+    //!\brief Befriend the base class so it can call impl().
+    friend base_t;
+
+    /*!\brief       Call the view's constructor with the underlying view as argument.
+     * \returns     An instance of seqan3::detail::view_drop.
```
Comment (rrahn) [inline include/seqan3/range/view/drop.hpp]: But you are returning a subrange

File: include/seqan3/range/view/view_all.hpp [hunk 34]
```diff
@@ -0,0 +1,176 @@
...
+#include <seqan3/std/span>
+
+namespace seqan3::detail
+{
+
+// ============================================================================
+//  all_fn (adaptor definition)
+// ============================================================================
+
+/*!\brief View adaptor definition for view::persist.
```
Comment (rrahn) [inline include/seqan3/range/view/view_all.hpp]: view::persist?

File: test/performance/range/view/view_all_benchmark.cpp [hunk 35]
```diff
@@ -0,0 +1,82 @@
...
+#include <string>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/range/view/view_all.hpp>
+
+using namespace seqan3;
+
+
```
Comment (rrahn) [inline test/performance/range/view/view_all_benchmark.cpp]: double new line

File: test/performance/range/view/view_drop_benchmark.cpp [hunk 36]
```diff
@@ -0,0 +1,156 @@
...
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/range/view/drop.hpp>
+#include <seqan3/range/view/single_pass_input.hpp>
+
+using namespace seqan3;
+
+
```
Comment (rrahn) [inline test/performance/range/view/view_drop_benchmark.cpp]: new line
